awsutil/awsutil.go

```diff
@@ -60,3 +71,22 @@ func UserAgentHandler(caller string) request.NamedHandler {
 		},
 	}
 }
+
+// NewSession prepares a client session.
+// The returned session includes a User-Agent handler to enable AWS to track usage.
+// If the AWS SDK fails to find the region, the region is parsed from Task metadata
+// (on EC2 the region is not typically defined in the environment).
+func NewSession(meta ECSTaskMeta, userAgentCaller string) (*session.Session, error) {
+	clientSession, err := session.NewSession()
+	if err != nil {
+		return clientSession, err
```

Comment:
```suggestion
		return nil, err
```

Reply:
nit: I know it doesn't really matter since go convention says don't use anything if err != nil but makes it a bit clearer it's not to be relied on.